data_steward/utils/bq.py

```diff
@@ -193,6 +193,50 @@ def define_dataset(project_id, dataset_id, description, label_or_tag):
     return dataset
 
 
+def update_labels_and_tags(project_id,
+                           dataset_id,
+                           label_or_tag,
+                           set_or_update=True):
+    """
+    Updates labels or tags in dataset if not set or needing to be updated
+    or overwrites existing labels or tags in the dataset
+
+    :param project_id:  string name to identify the project
+    :param dataset_id:  string name to identify the dataset
+    :param label_or_tag:  labels for the dataset = Dict[str, str]
+                          tags for the dataset = Dict[str, '']
+    :param set_or_update:  flag to signal if label_or_tag is to be either
+                             set or updated (True as default) or overwritten (False)
+
+    :raises:  RuntimeError if parameters are not specified
+
+    :return:  a dataset reference object
+    """
+    if not project_id:
+        raise RuntimeError(
+            "Specify the project_id for the project containing the dataset")
+
+    if not dataset_id:
+        raise RuntimeError("Provide a dataset_id")
+
+    if not label_or_tag:
+        raise RuntimeError("Please provide a label or tag")
+
+    client = get_client(project_id)
+    dataset_id = f"{project_id}.{dataset_id}"
+    dataset = client.get_dataset(dataset_id)
+
+    if set_or_update is True:
+        # will set or update labels and/or tags
+        dataset.labels = label_or_tag
+        dataset = client.update_dataset(dataset, ['labels'])
+        return dataset
+    if set_or_update is False:
```

Comment:
(See my comment above about this param vs `overwrite_ok`.)

Reply:
I agree that `overwrite_ok` is a better param name